DashboardTests/DemoTest.cs

```diff
@@ -87,5 +86,51 @@ public void TestDataStore()
             Assert.That(steeringAngle.Text, Is.EqualTo("Steering Angle: 90"));
             Assert.That(brakePressure.Text, Is.EqualTo("Brake Pressure: 100"));
         });
+
+    }
+
+    [AvaloniaTest]
+
+    public void TestNavigation()
+    {
+        // Arrange
+        var window = new MainWindowView()
+        {
+            DataContext = new MainWindowViewModel()
+        };
+
+        // Act
+        window.Show();
+
+        // Assert
+        Assert.That(window, Is.Not.Null);
+        Assert.That(window, Is.InstanceOf<MainWindowView>());
+
+        var naviBar = window.FindControl<ListBox>("NaviBar");
+        var mainContent = window.FindControl<ContentControl>("MainContent");
+        var defaultSelectedItem = new ListItemTemplate(typeof(ConnectionView), null, "Connection");
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(naviBar, Is.Not.Null);
+            Assert.That(mainContent, Is.Not.Null);
+        });
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(naviBar.SelectedItem, Is.EqualTo(defaultSelectedItem));
+            Assert.That(mainContent.Content, Is.InstanceOf(defaultSelectedItem.View));
+        });
+
+        var changedSelectedItem = new ListItemTemplate(typeof(StatusView), null, "Status");
```

Comment:
Thanks for adding testing, looks good!

Can you adjust / add a test that actually interacts with the UI to change the page? e.g. clicks a menu item and ensures it changed / also ensure some text or anything renders

Reply:
I can have a look but I don't know if there is functionality to simulate clicks.

Reply:
Nevermind my last, I have adjusted the TestNavigation test case to simulate mouse clicks.
https://github.com/nickowen44/AV-Datalogger/pull/6/commits/de242b75c4698bf8ec380ef2002a72363cba1937